phylogenetic/rules/prepare_sequences.smk

```diff
@@ -52,11 +52,11 @@ rule filter:
       - excluding strains with missing region, country or date metadata
     """
     input:
-        sequences = "data/sequences_{serotype}.fasta",
+        sequences = "data/{gene}/sequences_{serotype}.fasta",
```

Comment:
on L38 you write an output to `results/genome/filtered_{serotype}.fasta`.

here, on L59, you're writing to `results/{gene}/filtered_{serotype}.fasta` ...but I think (assuming I'm following this correctly) one of the values that gets bound to `gene` will be `"genome"`, which will then overwrite the file from L38 — am I confused, is that intentional?

Reply:
Ah L38 is writing output to `data/genome/sequences_{serotype}.fasta`

Here, I expect this rule to be able to write output to:
* `results/genome/filtered_{serotype}.fasta` <- currently works
* `results/E/filtered_{serotype}.fasta` <- will be added later 

Thanks for tracing input/output files through this though! Another way to check if any rules will overwrite the same files is by using the `-np` flag to dryrun the workflow, for example:

```bash
# from the dengue repo folder
nextstrain build phylogenetic results/genome/filtered_denv1.fasta -np
```

During a dryrun, if two rules are generating the same file(s) an `AmbiguousRuleException` is thrown and is usually [handled like this](https://snakemake.readthedocs.io/en/stable/snakefiles/rules.html#handling-ambiguous-rules) based on Snakemake docs.

Reply:
Thanks -- I think I was confused between L55 and L59, thanks for bearing with me. 

(I should probably sit down and read some Snakemake documentation too...)